private/buf/cmd/buf/command/alpha/plugin/pluginpush/pluginpush.go

```diff
@@ -249,3 +245,21 @@ func run(
 	}
 	return bufprint.NewCuratedPluginPrinter(container.Stdout()).PrintCuratedPlugin(ctx, format, curatedPlugin)
 }
+
+func loadDockerfile(ctx context.Context, bucket storage.ReadBucket) (storage.ReadObjectCloser, error) {
+	var err error
+	for _, path := range []string{bufplugindocker.SourceFilePath, bufplugindocker.SourceFileAlternatePath} {
+		var dockerfile storage.ReadObjectCloser
+		if dockerfile, err = bucket.Get(ctx, path); err == nil {
+			return dockerfile, err
+		}
+	}
+	if storage.IsNotExist(err) {
```

Comment:
The previous check for not exists was incorrect - we have to use the storage defined helper methods.

Reply:
Nice catch!